packages/execution-environments/src/common/BaseSnapExecutor.test.ts

```diff
@@ -708,6 +727,50 @@ describe('BaseSnapExecutor', () => {
     });
   });
 
+  it('supports onTransactionInsight export', async () => {
+    const CODE = `
+      module.exports.onTransactionInsight = ({ origin, transaction, chainId, metadata }) => ({ origin, transaction, chainId, metadata });
+    `;
+    const executor = new TestSnapExecutor();
+
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'executeSnap',
+      params: [FAKE_SNAP_NAME, CODE, []],
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    // @todo Updating the value to be closer to the value we expect from the extension.
+    // We also have to decide on the shape of that object.
+    const transaction = { maxFeePerGas: '0x' };
+
+    const params = { transaction, metadata: {}, chainId: '0x1' };
```

Comment:
The chain id is meant to be a universally unique [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#test-cases) id. Might be better to put `chainId: 'eip155:1'` here to avoid confusion in the future